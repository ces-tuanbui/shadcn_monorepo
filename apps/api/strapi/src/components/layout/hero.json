{
  "collectionName": "components_layout_heroes",
  "info": {
    "displayName": "Hero",
    "description": ""
  },
  "options": {},
  "attributes": {
    "heading": {
      "type": "string"
    },
    "text": {
      "type": "string"
    },
    "topLink": {
      "type": "component",
      "repeatable": false,
      "component": "elements.link"
    },
    "buttonLink": {
      "type": "component",
      "repeatable": true,
      "component": "elements.link"
    },
    "image": {
      "type": "media",
      "multiple": false,
      "required": false,
      "allowedTypes": [
        "images",
        "files",
        "videos",
        "audios"
      ]
    }
  }
}
