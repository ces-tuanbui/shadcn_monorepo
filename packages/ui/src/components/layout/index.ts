import { ThemeToggle } from "../layout/theme-toggle";
import { Header } from "../layout/header";
import { Footer } from "../layout/footer";

export { ThemeToggle, Header, Footer };
