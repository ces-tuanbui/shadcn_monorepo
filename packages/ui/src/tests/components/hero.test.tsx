import { render } from "@testing-library/react";
import { HeroSection } from "../../components/hero";

describe("HeroSection", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<HeroSection />);

    expect(asFragment()).toMatchSnapshot();
  });
});
