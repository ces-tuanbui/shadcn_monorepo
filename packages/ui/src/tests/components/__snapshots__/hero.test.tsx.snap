// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="container w-full mx-auto"
  >
    <div
      class="grid place-items-center lg:max-w-screen-xl gap-8 mx-auto py-20 md:py-32"
    >
      <div
        class="text-center space-y-8"
      >
        <div
          class="inline-flex items-center rounded-full border border-secondary px-2.5 font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground text-sm py-2"
        >
          <span
            class="mr-2 text-primary"
          >
            <div
              class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground hover:bg-primary/80"
            >
              New
            </div>
          </span>
          <span>
             Design is out now! 
          </span>
        </div>
        <div
          class="max-w-screen-md mx-auto text-center text-4xl md:text-6xl font-bold"
        >
          <h1>
            Experience the
            <span
              class="text-transparent px-2 bg-gradient-to-r from-[#D247BF] to-primary bg-clip-text"
            >
              Shadcn
            </span>
            landing page
          </h1>
        </div>
        <p
          class="max-w-screen-sm mx-auto text-xl text-muted-foreground"
        >
          We're more than just a tool, we're a community of passionate
            creators. Get access to exclusive resources, tutorials, and support.
        </p>
        <div
          class="space-y-4 md:space-y-0 md:space-x-4"
        >
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-5/6 md:w-1/4 font-bold group/arrow"
          >
            Get Started
            <svg
              class="lucide lucide-arrow-right size-5 ml-2 group-hover/arrow:translate-x-1 transition-transform"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12h14"
              />
              <path
                d="m12 5 7 7-7 7"
              />
            </svg>
          </button>
          <a
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2 w-5/6 md:w-1/4 font-bold"
            href="https://github.com/nobruf/shadcn-landing-page.git"
            target="_blank"
          >
            Github respository
          </a>
        </div>
      </div>
      <div
        class="relative group mt-14"
      >
        <div
          class="absolute top-2 lg:-top-8 left-1/2 transform -translate-x-1/2 w-[90%] mx-auto h-24 lg:h-80 bg-primary/50 rounded-full blur-3xl"
        />
        <img
          alt="dashboard"
          class="w-full md:w-[1200px] mx-auto rounded-lg relative rouded-lg leading-none flex items-center border border-t-2 border-secondary  border-t-primary/30"
          data-nimg="1"
          decoding="async"
          height="1200"
          loading="lazy"
          src="/_next/image?url=%2Fhero-image-dark.jpeg&w=3840&q=75"
          srcset="/_next/image?url=%2Fhero-image-dark.jpeg&w=1200&q=75 1x, /_next/image?url=%2Fhero-image-dark.jpeg&w=3840&q=75 2x"
          style="color: transparent;"
          width="1200"
        />
        <div
          class="absolute bottom-0 left-0 w-full h-20 md:h-28 bg-gradient-to-b from-background/0 via-background/50 to-background rounded-lg"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
